src/host/PtySignalInputThread.cpp

```diff
@@ -96,17 +96,29 @@ void PtySignalInputThread::CreatePseudoWindow()
 // Return Value:
 // - S_OK if the thread runs to completion.
 // - Otherwise it may cause an application termination and never return.
-[[nodiscard]] HRESULT PtySignalInputThread::_InputThread()
+[[nodiscard]] HRESULT PtySignalInputThread::_InputThread() noexcept
 {
-    PtySignal signalId;
-    while (_GetData(&signalId, sizeof(signalId)))
+    const auto shutdown = wil::scope_exit([this]() {
+        _Shutdown();
+    });
+
+    for (;;)
     {
+        PtySignal signalId;
+        if (!_GetData(&signalId, sizeof(signalId)))
+        {
+            return S_OK;
+        }

```

Comment:
I changed this in the end, so that the `_GetData` calls are all uniform and consistent. I just felt that this looks better...